arci/src/traits/motor_drive.rs

```diff
@@ -0,0 +1,28 @@
+use auto_impl::auto_impl;
+
+use crate::error::Error;
+
+#[auto_impl(Box, Arc)]
+pub trait MotorDrivePosition: Send + Sync {
+    fn set_motor_position(&self, actuator_state: ActuatorState) -> Result<(), Error>;
+    fn get_motor_position(&self) -> Result<Option<ActuatorState>, Error>;
+}
+
+#[auto_impl(Box, Arc)]
+pub trait MotorDriveVelocity: Send + Sync {
+    fn set_motor_velocity(&self, actuator_state: ActuatorState) -> Result<(), Error>;
+    fn get_motor_velocity(&self) -> Result<Option<ActuatorState>, Error>;
+}
+
+#[auto_impl(Box, Arc)]
+pub trait MotorDriveEffort: Send + Sync {
+    fn set_motor_effort(&self, actuator_state: ActuatorState) -> Result<(), Error>;
+    fn get_motor_effort(&self) -> Result<Option<ActuatorState>, Error>;
+}
+
+pub enum ActuatorState {

```

Comment:
We should not share this with among position/velocity/torque.
It's better to use different types to avoid some bugs.
I feel use f64 directly is enough...